hal/armv7a/zynq7000/hal.c

```diff
@@ -190,6 +190,35 @@ void hal_memcpy(void *dst, const void *src, unsigned int l)
 }
 
 
+void hal_memset(void *dst, int v, unsigned int l)
```

Comment:
If you are interested  I can provide optimized version of the `hal_memset()` function ( smaler size and uses fewer registers ).

Reply:
This code comes from [phoenix-rtos-kernel/hal/armv7a](https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/master/hal/armv7a/string.h).
If you would like to optimize it, go ahead. However, I would like to ask you to do it in separate PR associated with phoenix-rtos-kernel. This thread hones in on devs module so I wouldn't like to discuss others, not related topics here.

Reply:
I proposed this almost a year ago PR [#88](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/88)

Reply:
I will test it and add in a separate PR.